### ISSUE TITLE
AMQP Plugin using incorrect metric names

### ISSUE THREAD

**Issue by claytonpeters** [collectd/collectd#2063]
*   Version of collectd: 5.6.0 / 5.6.1
*   Operating system / distribution: Red Hat Enterprise Linux 7 (tested on both collectd from EPEL RPMs, and a compiled-from-source v5.6.1)

## Expected behavior

AMQP plugin sends metrics to RabbitMQ / Carbon-Cache with correct names

## Actual behavior

AMQP plugin sends some metrics on with the correct names, but also seems to add on metrics with the start of the name missing. For example:

```
<Plugin "amqp">
  <Publish "graphite">
    Host "192.168.169.177"
    Port "5672"
    VHost "/"
    User "<redacted>"
    Password "<redacted>"
    Format "Graphite"
    Exchange "metrics"
    GraphitePrefix "collectd."
    GraphiteEscapeChar "_"
    Persistent true
    StoreRates true
    ConnectionRetryDelay 0
  </Publish>
</Plugin>
```

With this configuration I get some correct metrics on my Graphite server, e.g. ```collectd.test01102_soton_ac_uk.load.load.midterm``` but I will also arbitrarily get, things like:
```
est01102_soton_ac_uk.load.load.midterm
on_ac_uk.load.load.midterm
idterm
```

These datasets will only sporadically get values rather than consistently.

From a packet capture it looks like a message is being sent with another previous message appearing at the end of the message, as if a buffer has been re-used.

This bug is not present in collectd 5.5.0 (compiled from source), but is present in both 5.6.0, 5.6.1 (compiled from source) and in the packages for RHEL 7 supplied by EPEL (which are based on 5.6.0)

## Steps to reproduce

* Install collectd collectd-amqp from EPEL or compile v5.6.1 from source
* Configure AMQP plugin along with some other collection plugins (load, cpu, disk, etc.) to send metrics
* See the metrics that it creates are sometimes missing the beginning of their names.

## Fix:

Having taken a look at recent changes to the source, I noted that in amqp.c in the ```camqp_write``` function, there was previously a ```memset()``` clearing out the ```buffer``` variable. In two commits in August (https://github.com/collectd/collectd/commit/69b8a9a1af204685dfdfaf1279a0f2928e0bbb32 and https://github.com/collectd/collectd/commit/baa2269a17f44cf8bb10bd57f9739b6e286aca5a), this was replaced with a ```char buffer[8192] = {0}``` initialiser, which was then removed on the belief that all the messages added to this buffer were null-terminated.

Adding ```memset (buffer, 0, sizeof (buffer));``` back in to the source seems to fix the issue.

**Comment by rubenk** [comment]
Thanks for the report @claytonpeters, and sorry for the breakage.
If you revert baa2269 locally, is that enough?

**Comment by claytonpeters** [comment]
I'll give that a go and get back to you shortly.

**Comment by claytonpeters** [comment]
Ok, it looks like it's been fixed between 5.6.1 and the latest in GitHub. It looked like it got fixed in commit dcd0b4f3121d903070282ee013b2de939bc0fa49 on Oct 11th (so a few days after the release of 5.6.1) by adding a missing null-terminator in utils_format_graphite.c

As such, it's already sorted and will be fixed whenever 5.6.2 comes out!

Apologies for not testing with the latest source before raising the bug!
